### PR TITLE
Add CupertinoAlertDialog

### LINKED ISSUE #7375: Implement CupertinoAlertDialog
https://developer.apple.com/ios/human-interface-guidelines/ui-views/alerts/

### PR DESCRIPTION
**Pull request by abarth**
Fixes #7375

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/dialog.dart [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+    children.add(new SizedBox(height: 20.0));
+
+    if (actions != null) {
+      children.add(new _CupertinoButtonBar(
+        children: actions,
+      ));
+    }
+
+    return new CupertinoDialog(
+      child: new IntrinsicWidth(
```
Comment (Hixie) [inline packages/flutter/lib/src/cupertino/dialog.dart]: we should remove this if it doesn't do anything

File: packages/flutter/lib/src/cupertino/dialog.dart [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+  bool get enabled => onPressed != null;
+
+  @override
+  Widget build(BuildContext context) {
+    TextStyle style = _kCupertinoDialogActionStyle;
+
+    if (isDestructive)
+      style = style.copyWith(color: _kDestructiveActionColor);
+
+    if (enabled)
```
Comment (Hixie) [inline packages/flutter/lib/src/cupertino/dialog.dart]: backwards

Comment (Hixie) [inline packages/flutter/lib/src/cupertino/dialog.dart]: we should have a test that CupertinoDialogAction picks the right colours and styles in the face of the various options

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/cupertino/dialog.dart (hunk 2 above)
Comment (abarth) [inline packages/flutter/lib/src/cupertino/dialog.dart]: This test actually caught a bug, btw.  We were computing the `style` and then ignoring it.

Comment (Hixie) [inline packages/flutter/lib/src/cupertino/dialog.dart]: haha nice. Write Test Find Bug!

**Pull request merged**